### ISSUE TITLE
Set GPU request and limit

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#2417]
W the k8s 1.8 release we have to set both. 

Right now we set the `alpha.kubernetes.io/nvidia-gpu` to 0 even for pipelines that don't specify a GPU resource. K8s 1.8 does not like that. We get an error like:

```
ReplicationController \"pipeline-xxx\" is invalid: spec.template.spec.containers[0].resources.requests: Invalid value: \"0\": must be equal to alpha.kubernetes.io/nvidia-gpu request","max_queue_size":10}]}}
```

Which is bad. Basically it means we're not compatible w k8s 1.8 yet.

**Comment by jdoliner** [comment]
I think we only want to set the limits section, according to: https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#api

**Comment by Czxck001** [comment]
@jdoliner I found this issue helps when I tried to walk through the official tutorials.
It seems currently released Pachyderm has not yet perfectly been compatible with k8s 1.8. Unfortunately in official tutorial I failed to see anything about this compatibility issue was mentioned. I followed the tutorial guide and get kubectl 1.8 and minikube v0.23.0 installed, and later failed.
I suppose it would be better if the official [tutorial & documentation](http://pachyderm.readthedocs.io/en/latest/getting_started/local_installation.html) could be updated with this issue mentioned. i.e. suggest using k8s 1.7.10 and minikube v0.22.3 (which works on my Macbook)
